src/wp-includes/block-supports/typography.php

```diff
@@ -480,18 +480,25 @@ function wp_get_typography_font_size_value( $preset, $should_use_fluid_typograph
 
 	// Checks if fluid font sizes are activated.
 	$typography_settings         = wp_get_global_settings( array( 'typography' ) );
-	$should_use_fluid_typography = isset( $typography_settings['fluid'] ) && true === $typography_settings['fluid'] ? true : $should_use_fluid_typography;
+	$should_use_fluid_typography =
+		isset( $typography_settings['fluid'] ) &&
+		( true === $typography_settings['fluid'] || is_array( $typography_settings['fluid'] ) ) ?
+			true :
+			$should_use_fluid_typography;
```

Comment:
```suggestion
	if (
		isset( $typography_settings['fluid'] ) &&
		( true === $typography_settings['fluid'] || is_array( $typography_settings['fluid'] )
	) {
		$should_use_fluid_typography = true;
	}
```

This change eliminates the reassignment of `$should_use_fluid_typography` to itself, making the code more readable.